assets/src/domain/shared/services/utils/dateAndTime.ts

```diff
@@ -60,6 +72,33 @@ const transformTime = (value: any, originalValue: any): Date => {
 	return transformDateTime(value, originalValue, MOMENT_TIME_FORMAT);
 };
 
+/**
+ * sets year, month and date for a time value from the given date object.
+ */
+export const updateTimeFromDate = (time: Date, date: Date): Date => {
+	// clone to avoid mutating the actual object
```

Comment:
perfect

Reply:
not only it prevents obsolete re-renders, but this way some of the unexpected bugs are prevented :+1: